python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -208,8 +214,9 @@ def evaluate(self, *, cache: MutableMapping[int, DataFrame]) -> DataFrame:
             translation phase should fail earlier.
         """
         return self.do_evaluate(
+            config,
```

Comment:
@wence- - Just a note. Pretty sure this means we will need to pass in a config object to every single task in the task graph for multi-gpu.

Reply:
Argh, ok, painful. Let's try and figure something out (especially because the config object can contain a memory resource).

Reply:
Would it make sense for `config` to be a required IR constructor argument, and **not** require it as an argument to `do_evaluate` (unless necessary)?

Reply:
Ah, we could pass the config options we need into the `Scan` node during `translate`, and then it never needs to be in `do_evaluate` at all

Reply:
Possible revision proposed here: https://github.com/rapidsai/cudf/pull/17339